scripts/clang-tidy.sh

```diff
@@ -30,7 +30,15 @@ if [[ ! -f build/compile_commands.json ]]; then
     mkdir -p build
     # Run make, pipe the output to the generate_compile_commands.py
     # and drop them in a place that clang-tidy will automatically find them
-    make | scripts/generate_compile_commands.py > build/compile_commands.json
+    RESULT=0
+    make > /tmp/make-node-cpp-skel-build-output.txt || RESULT=$?
+    if [[ ${RESULT} != 0 ]]; then
+        echo "Build failed, could not generate compile commands for clang-tidy, aborting!"
+        exit ${RESULT}
+    else
+        cat /tmp/make-node-cpp-skel-build-output.txt | scripts/generate_compile_commands.py > build/compile_commands.json
+    fi
+
```

Comment:
This change fixes a potential problem that existed before: if the build failed it would leave behind an empty `build/compile_commands.json`. Then the script logic would see that and avoid running the build again, leading clang-tidy to report nothing.

Reply:
> wondering why you can't just make > build/compile_commands.json? This is probably a sign of my lack of bash knowledge.

The `scripts/generate_compile_commands.py` is needed to reformat the output into this specific format (https://clang.llvm.org/docs/JSONCompilationDatabase.html). Some build systems can do this automatically (like cmake https://cmake.org/cmake/help/v3.5/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html) but here we use `gyp`, which cannot. So the `scripts/generate_compile_commands.py` is my attempt to bolt this functionality onto gyp.

> So the second way looks better to me. It's easier to read make > /tmp/make-node-cpp-skel-build-output.txt || RESULT=$? since you just use redirection and || is pretty much universally understood cross languages (I believe).

Great, thanks for the feedback. Its more verbose, but I also find it more readible since it breaks down all the things that are happening (return codes, output, flow) more.